catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeMapImpl.java

```diff
@@ -100,6 +102,26 @@ public String getValue() {
     return Objects.toString(json.get(VALUE));
   }
 
+  @Override
+  public Double getDistance() {
+    if (!isLeaf()) {
+      throw new IllegalStateException("No target value exists for a logical operator");
+    }
+
+    Object distance = json.get(DISTANCE);
+    if (distance == null) {
+      return null;
+    } else if (distance instanceof Double) {
+      return (Double) distance;
+    } else if (distance instanceof Integer) {
+      return ((Integer) distance).doubleValue();
+    } else if (distance instanceof Long) {
+      return ((Long) distance).doubleValue();
+    } else {
+      throw new IllegalStateException("The distance value could not be converted into a double");
+    }
```

Comment:
❓ Do we need to support all three? I would of thought just `Double` would suffice.

Reply:
Yes, the distance value is parsed as an Integer or Long if a whole number is used on the front end. I'll try @bellcc's Number approach though